games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+    if (
+      this.players[0].hand.length < 1 ||
+      this.players[1].hand.length < 1
+    ) {
+      console.log('ゲーム終了');
+      await this.showTurnLog();
+    }
+
+    if (this.isPlayerTurn) {
+      console.log('player Turn !!!');
+      console.log(this.arrayCardsPutOnMeldZone);
+      await this.showTurnLog();
+
+      await this.playerTurn().then(() => {
+        this.isPlayerTurn = !this.isPlayerTurn;
+        this.startGame();
+      });
+    } else {
+      console.log('house Turn !!!');
+      console.log(this.arrayCardsPutOnMeldZone);
+      await this.showTurnLog();
+
+      await this.houseTurn().then(() => {
+        this.isPlayerTurn = !this.isPlayerTurn;
+        this.startGame();
+      });
+    }
+  }
+
+  // 手札作成関数
+  private createHands(
+    deck: Deck,
+    index: number,
+    player: RummyPlayer
+  ) {
+    if (player.playerType === 'player') {
+      console.log(index);
+      for (let i = 0; i < 10; i += 1) {
+        const currCard = deck.drawOne();
+        if (currCard) {
+          player.addCardToHand(currCard);
+          currCard.playMoveTween(
+            this.playerHandZones[index].x -
+              (GAME.CARD.WIDTH + 10) * 2 +
+              i * (GAME.CARD.WIDTH + 10) -
+              350,
+            this.playerHandZones[index].y + 50
+          );
+          currCard.playFlipOverTween();
+        }
+      }
+    } else {
+      console.log(index);
+      for (let i = 0; i < 10; i += 1) {
+        const currCard = deck.drawOne();
+        if (currCard) {
+          player.addCardToHand(currCard);
+          currCard.playMoveTween(
+            this.playerHandZones[index].x -
+              (GAME.CARD.WIDTH + 10) * 2 +
+              i * (GAME.CARD.WIDTH + 10) -
+              350,
+            this.playerHandZones[index].y - 50
+          );
+        }
+      }
+    }
+  }
+
+  // TODO: playerTurn と houseTurnの間隔を合わせる
+  private async showTurnLog() {
+    if (this.turnText) this.turnText.destroy();
+    if (
+      this.players[0].hand.length <= 0 ||
+      this.players[1].hand.length <= 0
+    ) {
+      const message = 'ゲーム終了!!!';
+      this.turnText = this.add.text(
+        this.cameras.main.centerX,
+        this.cameras.main.centerY,
+        message,
+        { fontSize: '100px', color: '#000' }
+      );
+
+      this.turnText.setOrigin(0.5);
+      return new Promise((resolve) => {
+        this.time.delayedCall(5000, resolve);
+      });
+    }
+
+    const message = this.isPlayerTurn
+      ? 'Player turn!!'
+      : 'House turn!!';
+
+    this.turnText = this.add.text(
+      this.cameras.main.centerX,
+      this.cameras.main.centerY,
+      message,
+      { fontSize: '100px', color: '#000' }
+    );
+
+    this.turnText.setOrigin(0.5);
+    return new Promise((resolve) => {
+      this.time.delayedCall(5000, resolve);
+    });
+  }
+
+  private async playerTurn() {
+    // TODO: player hand 2 playermeld 3 捨てる作業
+    console.log('playerTurn!!');
+    if (this.deckBack.length < 1) {
```

Comment:
this.deckBackを配列からDeckインスタンスに変更した前提です。
this.deckBack.isEmpty()を使いましょう。


```suggestion
    if (this.deckBack.isEmpty()) {
```